62.CAD/common.hlsl

```diff
@@ -33,28 +42,24 @@ struct DrawObject
 struct QuadraticBezierInfo
 {
     float64_t2 p[3]; // 16*3=48bytes
+    float64_t2 arcLen;
+};
+
+struct CurveBox 
+{
+    // will get transformed in the vertex shader, and will be calculated on the cpu when generating these boxes
+    float64_t2 aabbMin;
+    float64_t2 aabbMax; // 32
+    float64_t2 curveMin[3]; // 80
+    float64_t2 curveMax[3]; // 128
```

Comment:
we talked about these values which are in [0, 1] range, don't need to have double precision.
pack/unpack each into/from a uint32_t
so you'll have 3 `uint32_t2`'s for each curve.
```
// on cpu
static_cast<uint32_t>(curveMin[i] * nbl::hlsl::numeric_limits<uint32_t>::max);
// on gpu
curveMin[i] / static_cast<float>(nbl::hlsl::numeric_limits<uint32_t>::max);
``` 
0 maps to 0.0
UINT_MAX maps to 1.0

Reply:
by the way, 
D:/projetos/polonia/Nabla/examples_tests/62.CAD/vertex_shader.hlsl:7545:29: error: C++-style cast is unsupported in HLSL

Reply:
Ah yea, then to regular (uint) implicit cast